### PR TITLE
[Email] Add retention policy for email inbox

### PR DESCRIPTION
**Pull request by darjoo**
#### Summary 
Add retention policy for Email Inbox. The length being 2 days retention.

#### Work Item(s) 
Fixes [AB#539754](https://dynamicssmb2.visualstudio.com/1fcb79e7-ab07-432a-a3c6-6cf5a88ba4a5/_workitems/edit/539754)

### CODE REVIEW

File: src/System Application/App/Email/src/EmailInstaller.Codeunit.al [hunk 1]
```diff
@@ -46,21 +48,73 @@ codeunit 1596 "Email Installer"
...
+
+        RetentionPolicySetup.SetRange("Table Id", Database::"Email Inbox");
+        if not RetentionPolicySetup.IsEmpty() then
+            exit;
+
+        RetentionPolicySetup.Validate("Table Id", Database::"Email Inbox");
+        RetentionPolicySetup.Validate("Apply to all records", true);
+        RetentionPolicySetup.Validate("Retention Period", RetentionPolicySetupCU.FindOrCreateRetentionPeriod("Retention Period Enum"::"2 Days"));
+        RetentionPolicySetup.Validate(Enabled, true);
+        RetentionPolicySetup.Insert(true);
```
Comment (stkillen) [inline src/System Application/App/Email/src/EmailInstaller.Codeunit.al:95]: Wouldn't this fail on `ForceUpdate = true` if this is already inserted? It also seems it's only called with ForceUpdate false. Is this parameter needed?

Comment (darjoo) [reply src/System Application/App/Email/src/EmailInstaller.Codeunit.al:95]: No, it wouldn't fail. There's a check above that the record doesn't exist. It'd exit if it exists.

**Review by stkillen — DISMISSED** [review]

File: src/System Application/App/Retention Policy/src/Retention Period/RetentionPeriodEnum.Enum.al [hunk 2]
```diff
@@ -29,6 +29,14 @@ enum 3900 "Retention Period Enum" implements "Retention Period"
         Caption = 'Custom';
     }
     /// <Summary>
+    /// The "2 Days" value results in a retention period where records that are older than 2 days are deleted.
+    /// </Summary>
+    value(10; "2 Days")
```
Comment (grobyns) [inline src/System Application/App/Retention Policy/src/Retention Period/RetentionPeriodEnum.Enum.al]: do we need this? Why not use a custom date calc thing?

**Review by JesperSchulz — DISMISSED** [review]
I find two days extremely short, but ok :-)

**Review by grobyns — APPROVED** [review]

**Comment by darjoo** [comment]
> I find two days extremely short, but ok :-)

They can always disable or lengthen if needed.

**Pull request merged**